### PR TITLE
Optionally provide a StringIO for the login flow prompt

### PR DESCRIPTION
**Pull request by sminot**
Using this method, I am able to prompt the user for login directly inside a Streamlit app

https://github.com/CirroBio/Cirro-client/assets/5215560/8246b419-765b-4c59-adae-101eceb6f69c

### CODE REVIEW

File: cirro/sdk/portal.py [hunk 1]
```diff
@@ -11,15 +13,15 @@ class DataPortal:
     available in the Data Portal.
     """
 
-    def __init__(self, client: DataPortalClient = None):
+    def __init__(self, client: DataPortalClient = None, auth_io: Optional[StringIO] = None):
```
Comment (nathanthorpe) [inline cirro/sdk/portal.py:16]: just revert the changes to this file and it should be good.

**Review by nathanthorpe — APPROVED** [review]

**Comment by sminot** [comment]
I've confirmed that this works nicely after removing `auth_io=` from the `DataPortal` init. 

@nathanthorpe, I was thinking I'd bump to 0.7.3 and merge in. Sound good?
